packages/flutter/lib/src/widgets/scroll_delegate.dart

```diff
@@ -861,3 +862,230 @@ Widget _createErrorWidget(Object exception, StackTrace stackTrace) {
   FlutterError.reportError(details);
   return ErrorWidget.builder(details);
 }
+
+// TODO(Piinks): Come back and add keep alive support, <BUG>
+/// A delegate that supplies children for scrolling in two dimensions.
+///
+/// A [TwoDimensionalScrollView] lazily constructs its box children to avoid
+/// creating more children than are visible through the
+/// [TwoDimensionalViewport]. Rather than receiving children as an
+/// explicit [List], it receives its children using a
+/// [TwoDimensionalChildDelegate].
+abstract class TwoDimensionalChildDelegate extends ChangeNotifier {
+  /// Returns the child with the given [ChildVicinity], which is described in
+  /// terms of x and y indices.
+  ///
+  /// Subclasses typically override this function and wrap their children in
+  /// [RepaintBoundary] widgets.
+  ///
+  /// The values returned by this method are cached. To indicate that the
+  /// widgets have changed, a new delegate must be provided, and the new
+  /// delegate's [shouldRebuild] method must return true.
+  Widget? build(BuildContext context, ChildVicinity vicinity);
+
+  /// Called whenever a new instance of the child delegate class is
+  /// provided.
+  ///
+  /// If the new instance represents different information than the old
+  /// instance, then the method should return true, otherwise it should return
+  /// false.
+  ///
+  /// If the method returns false, then the [build] call might be optimized
+  /// away.
+  bool shouldRebuild(covariant TwoDimensionalChildDelegate oldDelegate);
+}
+
+/// A delegate that supplies children for a [TwoDimensionalScrollView] using a
+/// builder callback.
+///
+/// The widgets returned from the builder callback are automatically wrapped in
+/// [RepaintBoundary] widgets if [addRepaintBoundaries] is true
+/// (also the default).
+///
+/// See also:
+///
+///  * [TwoDimensionalChildListDelegate], which is a similar delegate that has an
+///    explicit list of children.
+abstract class TwoDimensionalChildBuilderDelegate extends TwoDimensionalChildDelegate {
+  /// Creates a delegate that supplies children for a [TwoDimensionalScrollView]
+  /// using the given builder callback.
+  ///
+  /// The [builder] and [addRepaintBoundaries] arguments must not be null.
+  TwoDimensionalChildBuilderDelegate({
+    this.addRepaintBoundaries = true,
+    required this.builder,
+    this.maxXIndex,
+    this.maxYIndex,
```

Comment:
I am not convinced these are necessary (also for the list delegate), but my thinking was to imitate `childCount` and make it available to subclasses... but they could also add it in their own implementation, or we could add it later if the need becomes more apparent. Not sure.

Reply:
What if we just make it concrete then? I think that is ok, in the context of the test implementation, I noticed there is no need to add more implementation.
I feel like I started with abstract classes based on all the extra stuff I added to the delegate for the table view and as it developed I never considered making it concrete. I will update to make it so.

Reply:
> What if we just make it concrete then?

Sounds reasonable.